tests/common-lib/index.js

```diff
@@ -3,6 +3,26 @@
 const { execSync } = require('child_process')
 const lodash = require('lodash')
 
+/**
+ * Compares the length of two arrays and determine if the
+ * @param {*} receieved The target array.
+ * @param {*} expected The source array.
+ * @param {*} range The accepted range of difference between the arrays.
+ * @returns
+ */
+function closeMatch(receieved, expected, range = 3) {
+  if (receieved.length === 0) {
+    return false
+  } else if (
+    receieved.length <= expected.length + range &&
+    receieved.length >= expected.length - range
+  ) {
+    return true
+  }
```

Comment:
There is a chance that search will have 0 resources and kube API will have 1. We will need to revisit this logic within a later PR.